agent/agent.go

```diff
@@ -1254,6 +1260,45 @@ func (a *Agent) clearGitConfig(tc *taskContext) {
 		return
 	}
 	logger.Info("Cleared git credentials.")
+
+	if err := unregisterScalar(); err != nil {
+		logger.Error(errors.Wrap(err, "unregistering scalar repositories"))
+		return
+	}
+}
+
+// unregisterScalar unregisters a repository that was cloned (and therefore registered with) Scalar.
+// This should be done for each repository after it is cloned with scalar to stop the background maintenance processes.
+func unregisterScalar() error {
+	isScalarAvailable, err := agentutil.IsGitVersionMinimum(thirdparty.RequiredScalarGitVersion)
+
+	if err != nil {
```

Comment:
small nit: I'm okay with it either way, but we could do `if err != nil || !isScalarAvailable` because `errors.Wrap` will return nil for `err = nil`